src/util/psbt/mod.rs

```diff
@@ -72,6 +72,28 @@ pub struct PartiallySignedTransaction {
 }
 
 impl PartiallySignedTransaction {
+
+    /// Returns a list of utxos
+    pub fn to_prevouts(&self) -> Result<Vec<&TxOut>, Error> {
+        let mut prev_outs : Vec<&TxOut> = Vec::new(); 
+        for i in 0..self.inputs.len() {
+            let psbt_input = &self.inputs[i];
+             // check for witness utxos
+                if let Some(ref witness_utxo) = psbt_input.witness_utxo {
+                    prev_outs.push(&witness_utxo)
+                     // if no witness utxos then catch the non witness part of the input
+                } else if let Some(ref non_witness_utxo) = psbt_input.non_witness_utxo {
+                     // get output index (vout) from the outpoint and use it to get the non_witness_utxo
+                    let vout = self.unsigned_tx.input[i].previous_output.vout as usize;
+                    prev_outs.push(&non_witness_utxo.output[vout])
+                } else {
+                    // created MissingUtxo error enum in errors.rs
+                    return Err(Error::MissingUtxo);
+                }
+            }
+            Ok(prev_outs)
```

Comment:
Definitely *not*, that'd defeat the pointof iterator which is to avoid allocation. The whole function must be one giant iterator - presumably `zip()`ing together inputs and mapping them.

Also I just realized the PSBT type is **not** guaranteed to have inputs of same length. The current implementation panics implicitly in **some** unequal-len situations which isn't documented. I think we should just add `assert_eq!()` at the top and document that the function panics if lengths are not equal. Long term we will change the API to statically guarantee same lengths. I believe @dr-orlovsky is working on that.

Reply:
Cool, cheers.

Reply:
@Kixunil and @tcharding , iterators are pretty cool but returning `impl Iterator<item=R<T, E>>` seems a little tricky, tried to run locally but needed itertools, [see this](https://users.rust-lang.org/t/like-early-return-on-iterator-error/27589/11), we don't want to install extra dependencies, is there a way to skip itertools here ?

Reply:
`zip()`ped psbt inputs and transaction inputs, [code](https://pastebin.com/FkzTKAa7), this one gives error on line12 (inside `else{}`), then used `itertools::process_results` which claimed to handle `Iterator<Item = Result<T, E>>` and it worked without error, but it required me to install an extra dependency, which I don't think is a good idea.....is there a better way ?
could have used filter_map() and partition() but former just ignored the errors and latter would partition Ok and Err but I want the code to throw error once the first error is detected.